Content.Server/GameObjects/Components/Items/RCDComponent.cs

```diff
@@ -0,0 +1,169 @@
+         * @params message = The original message from examining, like ..() in BYOND's examine
+         *
+         */
+
+        void IExamine.Examine(FormattedMessage message)
+        {
+            message.AddMarkup(Loc.GetString("It's currently in "+mode+" mode, and holds "+this.ammo+" charges."));
+        }
+
+        /**
+         *
+         * Method to handle clicking on a tile to then appropriately RCD it. This can have several behaviours depending on mode.
+         * @param eventAargs = An action event telling us what tile was clicked on. We use this to exrapolate where to place the new tile / remove the old one etc.
+         *
+         */
+
+        public void AfterAttack(AfterAttackEventArgs eventArgs)
+        {
+            var attacked = eventArgs.Attacked;
+            var mapGrid = _mapManager.GetGrid(eventArgs.ClickLocation.GridID);
+            var tile = mapGrid.GetTileRef(eventArgs.ClickLocation);
+
+            var coordinates = mapGrid.GridTileToLocal(tile.GridIndices);
+            float distance = coordinates.Distance(_mapManager, Owner.Transform.GridPosition);
+
+            if (distance > InteractionSystem.InteractionRange || ammo <= 0)
+            {
+                return;
+            }
+
+            var targetTile = (ContentTileDefinition) _tileDefinitionManager[tile.Tile.TypeId];
+            ITileDefinition desiredTile = null;
+            //Try catch to assert that _outputTile is a valid key in the tiledefinition manager, failing this check, we'll abort any tile placement.
+            try
+            {
+                desiredTile = _tileDefinitionManager[_outputTile];
+            }
+            catch (Exception e)
+            {
+                desiredTile = null;
+            }
+
+            var canPlaceTile = targetTile.IsSubFloor && desiredTile != null; //Boolean to check if we're able to build the desired tile. This defaults to checking for subfloors, but is overridden by "deconstruct" which sets it to the inverse.
+
+
+            switch (this.mode)
+            {
+                //Floor mode just needs the tile to be a space tile (subFloor)
+                case RCDmode.Floors:
+                    break;
+                //We don't want to place a space tile on something that's already a space tile. Let's do the inverse of the last check.
+                case RCDmode.Deconstruct:
+                    canPlaceTile = !targetTile.IsSubFloor && desiredTile != null;
+                    break;
+                //Walls are a special behaviour, and require us to build a new object with a transform rather than setting a grid tile, thus we early return to avoid the tile set code.
+                case RCDmode.Walls:
+                    //If for whatever reason we can't get the grid coords, something has gone horribly wrong so let's return.
+                    if (!_mapManager.TryGetGrid(eventArgs.ClickLocation.GridID, out var grid))
```

Comment:
Again, I don't see why this is necessary. Is the value you want not already defined as `mapGrid`?